src/com/seafile/seadroid2/cameraupload/MediaObserverService.java

```diff
@@ -0,0 +1,139 @@
+package com.seafile.seadroid2.cameraupload;
+
+import android.accounts.OnAccountsUpdateListener;
+import android.app.Service;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.database.ContentObserver;
+import android.os.IBinder;
+import android.provider.MediaStore;
+import android.util.Log;
+
+import com.seafile.seadroid2.SettingsManager;
+
+/**
+ * This service monitors the media provider content provider for new images/videos.
+ *
+ * If new content appears, this service will get notified and send a syncRequest to the MediaSyncProvider.
+ *
+ * This service is always running, even if camera upload is not active.
+ * However, it will only register it's ContentObservers if Camera Upload is enabled in Seadroid.
+ */
+public class MediaObserverService extends Service {
+    private static final String DEBUG_TAG = "MediaObserverService";
+
+    private MediaObserver mediaObserver = null;
+    private SettingsManager settingsManager = SettingsManager.instance();
+    private CameraUploadManager cameraManager;
+
+    /**
+     * If camera upload settings have changed, we might have to trigger a full resync.
+     * This listener takes care of that.
+     */
+    private SharedPreferences.OnSharedPreferenceChangeListener settingsListener =
+            new SharedPreferences.OnSharedPreferenceChangeListener() {
+
+        @Override
+        public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
+
+            boolean doFullResync = false;
+
+            // here we have to catch *all* the cases that might make a full resync to the repository
+            // necessary.
+
+            switch (key) {
+
+                // if video upload has been switched on, do a full sync, to upload
+                // any older videos already recorded.
+                case SettingsManager.CAMERA_UPLOAD_ALLOW_VIDEOS_SWITCH_KEY:
+                    if (settingsManager.isVideosUploadAllowed())
+                        doFullResync = true;
+                    break;
+
+                // same goes for if the list of selected buckets has been changed
+                case SettingsManager.SHARED_PREF_CAMERA_UPLOAD_BUCKETS:
+                    doFullResync = true;
+                    break;
+
+                // the repo changed, also do a full resync
+                case SettingsManager.SHARED_PREF_CAMERA_UPLOAD_REPO_ID:
+                    doFullResync = true;
+                    break;
+
+                // the repo dir, also do a full resync
+                case SettingsManager.SHARED_PREF_CAMERA_UPLOAD_REPO_DIR:
+                    doFullResync = true;
+                    break;
+            }
+
+            if (cameraManager.isCameraUploadEnabled() && doFullResync) {
```

Comment:
Alright, I have noticed that https://github.com/forouher/seadroid/blob/refactor/account_and_camera_upload/src/com/seafile/seadroid2/cameraupload/CameraSyncAdapter.java#l267 already checked the network connections.